actions/actions.py

```diff
@@ -33,22 +34,35 @@ def etcdctl_path():
     return 'etcdctl'
 
 
-def etcdctl(cmd, etcdctl_api='3', endpoints=':4001', **kw):
+def etcdctl(cmd, etcdctl_api='3', endpoints=None, **kw):
     '''Call etcdctl with ``cmd`` as the command-line args.
 
     '''
     opts = layer.options('tls-client')
-
     env = os.environ.copy()
-    env['ETCDCTL_CA_FILE'] = opts['ca_certificate_path']
-    env['ETCDCTL_CERT_FILE'] = opts['server_certificate_path']
-    env['ETCDCTL_KEY_FILE'] = opts['server_key_path']
+
     if etcdctl_api:
         env['ETCDCTL_API'] = etcdctl_api
 
     etcdctl_cmd = etcdctl_path()
-    if endpoints:
-        etcdctl_cmd += ' --endpoints={}'.format(endpoints)
+
+    if endpoints is not False:
+        major, minor, _ = etcdctl_version().split('.')
+        if int(major) >= 3 and int(minor) >= 3:
+            env['ETCDCTL_CACERT'] = opts['ca_certificate_path']
+            env['ETCDCTL_CERT'] = opts['server_certificate_path']
+            env['ETCDCTL_KEY'] = opts['server_key_path']
+            if endpoints is None:
+                leader_address = leader_get('leader_address')
+                endpoints = leader_address
+                etcdctl_cmd += ' --endpoints={}'.format(endpoints)
```

Comment:
Can we use `http://127.0.0.1:4001` as the endpoint here?

Reply:
Yes, you're right.  I must have been mixing https and 4001 in testing.